File: RPICommunicator_V1/app/src/main/java/com/example/rpicommunicator_v1/Database/PlantDao.java
package com.example.rpicommunicator_v1.Database;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface PlantDao {

    @Insert
    void insert(Plant plant);

    @Update
    void update(Plant plant);

    @Delete
    void delete(Plant plant);

    @Query("SELECT * FROM plant_table")
    LiveData<List<Plant>> getAllPlants();//Wegen den Livedata als returntyp werden werte in der Liste immer automatisch updated


}

File: app/src/main/java/com/example/rpicommunicator_v1/database/plant/PlantDao.kt
package com.example.rpicommunicator_v1.database.plant

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface PlantDao {
    @Insert
    fun insert(plant: Plant?)

    @Update
    fun update(plant: Plant?)

    @Delete
    fun delete(plant: Plant?)

    @get:Query("SELECT * FROM plant_table")
    val allPlants: LiveData<List<Plant>>
}